src/intrinsic/mod.rs

```diff
@@ -1076,11 +1098,11 @@ impl<'a, 'gcc, 'tcx> Builder<'a, 'gcc, 'tcx> {
             res.to_rvalue()
         }
         else {
-            let res = lhs - rhs;
-            let comparison = self.context.new_comparison(None, ComparisonOp::LessThanEquals, res, lhs);
-            let comparison = self.context.new_cast(None, comparison, lhs.get_type());
-            let unary_op = self.context.new_unary_op(None, UnaryOp::Minus, comparison.get_type(), comparison);
-            self.and(res, unary_op)
+            assert!(!signed);
```

Comment:
This assert is not needed anymore since you reverted the above condition to `if signed`.

Reply:
Addressed in 5061e3ad163a75e81ed7b69e2113b5db135bc51a